presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2286,41 +2287,90 @@ public String toString()
     private static class PartitionAndMore
     {
         private final HiveIdentity identity;
-        private final Partition partition;
-        private final Path currentLocation;
-        private final Optional<List<String>> fileNames;
-        private final PartitionStatistics statistics;
-        private final PartitionStatistics statisticsUpdate;
 
-        public PartitionAndMore(HiveIdentity identity, Partition partition, Path currentLocation, Optional<List<String>> fileNames, PartitionStatistics statistics, PartitionStatistics statisticsUpdate)
+        public PartitionAndMore(HiveIdentity identity)
```

Comment:
Now the `PartitionAndMore` class doesn't have partition and doesn't have anything more.
I see it's used for inheritance, but i'd inline it.

Reply:
It's needed because partition manipulation actions are held together in ```@GuardedBy("this")
    private final Map<SchemaTableName, Map<List<String>, Action<PartitionAndMore>>> partitionActions = new HashMap<>()``` so either I will introduce interface, common super class or change partitionsActions to use ```Action<?>```